server/core/src/rodsAgent.cpp

```diff
@@ -250,14 +250,13 @@ int initRsCommFromServerSocket( rsComm_t *rsComm, int socket ) {
     }
 
     int newSocket;
-    receiveSocketFromSocket( socket, &newSocket );
-    rsComm->sock = newSocket;
-
-    if ( rsComm->sock != 0 ) {
-
-        /* remove error messages from xmsLog */
-        setLocalAddr( rsComm->sock, &rsComm->localAddr );
-        setRemoteAddr( rsComm->sock, &rsComm->remoteAddr );
+    if ( receiveSocketFromSocket( socket, &newSocket ) > 0 ) {
+        rsComm->sock = newSocket;
+        if ( 0 != rsComm->sock ) {
+            /* remove error messages from xmsLog */
+            setLocalAddr( rsComm->sock, &rsComm->localAddr );
+            setRemoteAddr( rsComm->sock, &rsComm->remoteAddr );
+        }
     }
```

Comment:
need an else() here?

Reply:
Upon closer inspection, looks like this function (initRsCommFromServerSocket) is not being referenced anywhere. We could fix it up or just remove it. Thoughts, @jasoncoposky?

Reply:
not used anywhere... torch it.